services/ui/ws/server_window.cc

```diff
@@ -177,7 +177,28 @@ void ServerWindow::SetBounds(
     const base::Optional<cc::LocalSurfaceId>& local_surface_id) {
   if (bounds_ == bounds && current_local_surface_id_ == local_surface_id)
     return;
+  SetBoundsInternal(bounds, local_surface_id);
+}
+
+void ServerWindow::SetBoundsFromHostWindowManager(
+    const gfx::Rect& bounds,
+    const base::Optional<cc::LocalSurfaceId>& local_surface_id) {
+  if (bounds_ == bounds && current_local_surface_id_ == local_surface_id)
+    return;
 
+  gfx::Rect old_bounds = bounds_;
+  SetBoundsInternal(bounds, local_surface_id);
+  if (bounds.origin() != old_bounds.origin()) {
+    // Setting another origin to |root_| must be avoided when bounds are sent
+    // from a window manager, but new bounds should be passed through
+    // OnWindowBoundsChanged.
+    bounds_.set_origin(old_bounds.origin());
```

Comment:
question: should not it be better in SetBoundsInternal we just set the size (not location), in line 202 below?